server/src/main/java/org/opensearch/index/SegmentReplicationRejectionStats.java

```diff
@@ -34,8 +34,7 @@ public SegmentReplicationRejectionStats(final long totalRejectionCount) {
     }
 
     public SegmentReplicationRejectionStats(StreamInput in) throws IOException {
-        // TODO: change to V_2_12_0 on main after backport to 2.x
-        if (in.getVersion().onOrAfter(Version.V_3_0_0)) {
+        if (in.getVersion().onOrAfter(Version.V_2_12_0)) {
```

Comment:
didn't catch this before but we don't need to add the wire compat here given the compat is handled in nodestats.